src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -285,86 +421,92 @@ def find_fixed_range(testcase_id, job_type):
-    raise errors.BuildNotFoundError(max_revision, job_type)
+    error_message = f'Build {max_revision} no longer exists.'
+    return uworker_io.UworkerOutput(
+        testcase=testcase,
+        uworker_input=uworker_input,
+        error_message=error_message,
+        error=uworker_msg_pb2.ErrorType.PROGRESSION_BUILD_NOT_FOUND)
 
   testcase = data_handler.get_testcase_by_id(testcase_id)
-  data_handler.update_testcase_comment(testcase, data_types.TaskState.STARTED,
-                                       'r%d' % max_revision)
 
   # Check to see if this testcase is still crashing now. If it is, then just
   # bail out.
-  result = _testcase_reproduces_in_revision(
+  result, error = _testcase_reproduces_in_revision(
+      uworker_input,
       testcase,
       testcase_file_path,
       job_type,
       max_revision,
       update_metadata=True)
+  if error is not None:
+    return error
+
   if result.is_crash():
-    logs.log('Found crash with same signature on latest revision r%d.' %
-             max_revision)
+    logs.log(f'Found crash with same signature on latest'
+             f' revision r{max_revision}.')
     app_path = environment.get_value('APP_PATH')
     command = testcase_manager.get_command_line_for_application(
         testcase_file_path, app_path=app_path, needs_http=testcase.http_flag)
     symbolized_crash_stacktrace = result.get_stacktrace(symbolized=True)
     unsymbolized_crash_stacktrace = result.get_stacktrace(symbolized=False)
     stacktrace = utils.get_crash_stacktrace_output(
         command, symbolized_crash_stacktrace, unsymbolized_crash_stacktrace)
-    testcase = data_handler.get_testcase_by_id(testcase_id)
-    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
-        stacktrace)
-    data_handler.update_progression_completion_metadata(
-        testcase,
-        max_revision,
-        is_crash=True,
-        message='still crashes on latest revision r%s' % max_revision)
-
-    # Since we've verified that the test case is still crashing, clear out any
-    # metadata indicating potential flake from previous runs.
-    task_creation.mark_unreproducible_if_flaky(testcase, False)
 
-    # For chromium project, save latest crash information for later upload
-    # to chromecrash/.
     state = result.get_symbolized_data()
-    crash_uploader.save_crash_info_if_needed(testcase_id, max_revision,
-                                             job_type, state.crash_type,
-                                             state.crash_address, state.frames)
-    return
+    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
+        stacktrace)
 
+    error_message = f'Still crashes on latest revision r{max_revision}.'
+    progression_task_output = uworker_io.ProgressionTaskOutput(
+        crash_on_latest=True,
+        crash_type=state.crash_type,
+        crash_address=state.crash_address,
+        # TODO(alhijazi): Check how to properly serialize the frames.
```

Comment:
I had created a field in the ProgressionTaskOutput for `crash_frames`. I chose string as a type but I'm not sure it's the appropriate type. I wanted to check with you before landing this.

Reply:
I think these are StackFrameStructure.
1. I think you should convert these to a dictionary (`__dict__` method probably works)
2. make crash_frames a JSON field. 
3. Override ProgressionTaskOutput's save_rich_type method to include this (it's done elsewhere in the file, use inspiration)
```
if isinstance(value, (dict, list)):
      save_json_field(field, value)
      return
```
4. In postprocess, convert back from the dict to StackFrameStructure
5. Test this very thouroghly!!!

Reply:
`state.crash_type`, `state.crash_address`, `state.frames` are only used to create `serialized_crash_stack_frames` in `crash_uploader.save_crash_info_if_needed` using `crash_uploader.get_symbolized_stack_bytes`.
So instead of passing these three variables and going through the trouble of serializing/deserializing them. I modified `save_crash_info_if_needed` to take the serialized frames. We now call `get_symbolized_stack_bytes` in the utask and pass the result to the trusted worker using progression_task_output.
`get_symbolized_stack_bytes` returns `process_state.SerializeToString()` (a string).